src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -3477,4 +3477,36 @@ WindowID MessageHelper::ExtractWindowIdFromSmartObject(
   return mobile_apis::PredefinedWindows::DEFAULT_WINDOW;
 }
 
+void MessageHelper::RemoveEmptyMessageParams(
+    smart_objects::SmartObject& msg_params) {
+  if (msg_params.empty()) {
+    return;
+  }
+
+  for (const auto& item : msg_params.enumerate()) {
+    auto& param = msg_params[item];
+
+    if (param.empty()) {
+      msg_params.erase(item);
```

Comment:
Is it safe to erase this param, received by reference, during iteration? I guess that for loop may be not the best option for this to exclude possibility of undefined behavior there

Reply:
I iterate on a set of keys in a for loop and delete from the map.

Reply:
But it may be unsafe. I believe you should use iterators instead of for loop.
Please look at https://stackoverflow.com/questions/2874441/deleting-elements-from-stdset-while-iterating

Reply:
done